apps/frontend/src/app/5_pages/MarketMakingPage/components/AdjustAndDepositModal/AdjustAndDepositModal.tsx

```diff
@@ -52,35 +62,120 @@ export const AdjustAndDepositModal: FC<AdjustAndDepositModalProps> = ({
 }) => {
   const [adjustType, setAdjustType] = useState(AdjustType.Deposit);
   const [value, setValue, amount] = useWeiAmountInput('');
+  const [tokenPoolBalance, setTokenPoolBalance] = useState('0');
+  const { account } = useAccount();
+  const { onDeposit, onWithdraw } = useHandleMarketMaking(onClose);
+  const { balanceA, loadingA } = useGetUserInfo(pool);
+
+  const poolWeiAmount = useMemo(
+    () =>
+      decimalic(amount.toString())
+        .div(decimalic(balanceA.toString()).div(tokenPoolBalance))
+        .toNumber()
+        .toFixed(0),
+    [amount, balanceA, tokenPoolBalance],
+  );
+  const token = useMemo(() => pool.assetA, [pool.assetA]);
+  const isDeposit = useMemo(
+    () => adjustType === AdjustType.Deposit,
+    [adjustType],
+  );
   const [hasDisclaimerBeenChecked, setHasDisclaimerBeenChecked] =
     useState(false);
 
-  const handleMaxClick = useCallback(
-    () => setValue(maxBalance.toString()),
-    [setValue],
+  const { weiBalance: maxTokenToDepositAmount } = useMaxAssetBalance(
+    token,
+    defaultChainId,
+    GAS_LIMIT.MARKET_MAKING_ADD_LIQUIDITY,
```

Comment:
```suggestion
  const { balance: maxTokenToDepositAmount } = useMaxAssetBalance(
    token,
    defaultChainId,
    isDeposit? GAS_LIMIT.MARKET_MAKING_ADD_LIQUIDITY : GAS_LIMIT.MARKET_MAKING_REMOVE_LIQUIDITY,
```
`balance` is of type `Decimal` so you don't need to convert `balanceA` in `maxBalance`. Regarding the second change - you should choose gas limit based on the `AdjustType`.